modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -677,27 +686,10 @@ private Predicate _startsWith(
 		Object fieldName, Object fieldValue, long objectDefinitionId) {
 
 		if (fieldName.equals("keywords")) {
-			return _getColumn(
-				"id", objectDefinitionId
-			).in(
-				DSLQueryFactoryUtil.select(
-					AssetEntryTable.INSTANCE.classPK
-				).from(
-					AssetEntryTable.INSTANCE
-				).innerJoinON(
-					AssetEntries_AssetTagsTable.INSTANCE,
-					AssetEntryTable.INSTANCE.entryId.eq(
-						AssetEntries_AssetTagsTable.INSTANCE.entryId)
-				).innerJoinON(
-					AssetTagTable.INSTANCE,
-					AssetTagTable.INSTANCE.tagId.eq(
-						AssetEntries_AssetTagsTable.INSTANCE.tagId
-					).and(
-						AssetTagTable.INSTANCE.name.like(
-							fieldValue + StringPool.PERCENT)
-					)
-				)
-			);
+			return _getKeywordsPredicate(
+				objectDefinitionId,
```

Comment:
Maybe this is more personal taste, but I would prefer to have a `_startsWith` that receives a column and some information to return the `.like` than start using ifs because at least for me it is a clearer way of what to expect of that method.

For me, this kind of ifs are adding "magic" that can make confusing the expected behavior.
But, if it is ok for you, for me is ok to have it this way.

Reply:
See https://github.com/liferay-headless/liferay-portal/pull/1078/commits/f8067c92d1fde9a88d3c3c9600f27c1763c36833.

Thanks!